src/main/scala/com/mwz/sonar/scala/scapegoat/ScapegoatRulesRepository.scala

```diff
@@ -0,0 +1,50 @@
+package com.mwz.sonar.scala
+package scapegoat
+
+import inspections.{Level, ScapegoatInspection}
+
+import org.sonar.api.server.rule.RulesDefinition
+import org.sonar.api.rule.RuleStatus
+import org.sonar.api.rule.Severity
+import org.sonar.api.rules.RuleType
+
+/** Defines a rules repository for the Scapegoat inspections */
+class ScapegoatRulesRepository extends RulesDefinition {
+
+  /** Defines the rules in the repository */
+  override def define(context: RulesDefinition.Context): Unit = {
+    //crete an empty repository
+    val repository =
+      context
+        .createRepository(ScapegoatRulesRepository.RepositoryKey, Scala.LanguageKey)
+        .setName(ScapegoatRulesRepository.RepositoryName)
+
+    //register each scapegoat inspection as a repository rule
+    for (inspection <- ScapegoatInspection.AllScapegoatInspections) {
+      val rule = repository.createRule(inspection.id)
+      val ruleSeverity = ScapegoatRulesRepository.scapegoatLevelToRuleSeverity(inspection.defaultLevel)
+
+      rule.setInternalKey(inspection.id)
+      rule.setName(inspection.name)
+      rule.setMarkdownDescription(inspection.description)
+      rule.setActivatedByDefault(false) // scalastyle:ignore LiteralArguments
+      rule.setStatus(RuleStatus.READY)
+      rule.setSeverity(ruleSeverity)
+      rule.setType(RuleType.CODE_SMELL)
+    }
+
+    //save the repository
+    repository.done()
+  }
+}
+
+object ScapegoatRulesRepository {
+  private val RepositoryKey = "ScapegoatRepository"
+  private val RepositoryName = "Scapegoat Repository"
+
+  private def scapegoatLevelToRuleSeverity(level: Level): String = level match {
```

Comment:
I think it's quite subjective, but personally, I would be less aggressive and I'd map `Warning` to `MINOR` and `Error` to `MAJOR` and leave it up to the users to increase anything to CRITICAL or a BLOCKER level. `BLOCKER` is the highest severity, so we probably don't want to be using that by default. 

It looks like currently all of the rules that we have in the public instance of [Sonar](https://sonar.sonar-scala.com/coding_rules) have either `INFO`, `MINOR` or `MAJOR` severity.

Do you have any thoughts on this?

Reply:
Not really, thats why I asked if that mapping was ok.
The only thing is that I mapped `Error` to `Blocker`, because, _for me_, an **error** is something that has to be fixed ASAP. But you're right that it should be an user decision.